wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js

```diff
@@ -956,7 +975,7 @@
 
 					try {
 						// do the evaluation in global scope
-						var f = window.eval(toExecute);
+						var f = jQuery.globalEval(toExecute, {nonce: nonce});
```

Comment:
I've written a blog post about this functionality: https://wicketinaction.com/2013/02/replace-components-with-animation/

Reply:
This kind of a black magic. 

Why can't this be done with something like this
```
function myHideFunction(id) {
  var el = $("#"+id);
  el.attr("id", id+"-anim").fadeOut(500);
  $("<span/>").attr("id", id).appendTo(el.parent());
}

function myShowFunction(id) {
  setTimeout(function() {
    var el = $("#"+id);
    el.fadeIn(500);
  }, 500);
}
```

or maybe
```
function mySubscriber(evt, id) {
  $("#"+id).fadeIn(500);
  Wicket.Event.unsubscribe("/myAnim/hidden/"+id, mySubscriber);
}

function myHideFunction(id) {
  var el = $("#"+id);
  el.attr("id", id+"-anim").fadeOut(500, function(){
    Wicket.Event.publish("/myAnim/hidden/"+id, id);
  });
  $("<span/>").attr("id", id).appendTo(el.parent());
}

function myShowFunction(id) {
  Wicket.Event.subscribe("/myAnim/hidden/"+id, mySubscriber);
}
```

Reply:
@martin-g This feels like kind of a black magic. 

Why can't this be done with something like this
```
function myHideFunction(id) {
  var el = $("#"+id);
  el.attr("id", id+"-anim").fadeOut(500);
  $("<span/>").attr("id", id).appendTo(el.parent());
}

function myShowFunction(id) {
  setTimeout(function() {
    var el = $("#"+id);
    el.fadeIn(500);
  }, 500);
}
```

or maybe
```
function mySubscriber(evt, id) {
  $("#"+id).fadeIn(500);
  Wicket.Event.unsubscribe("/myAnim/hidden/"+id, mySubscriber);
}

function myHideFunction(id) {
  var el = $("#"+id);
  el.attr("id", id+"-anim").fadeOut(500, function(){
    Wicket.Event.publish("/myAnim/hidden/"+id, id);
  });
  $("<span/>").attr("id", id).appendTo(el.parent());
}

function myShowFunction(id) {
  Wicket.Event.subscribe("/myAnim/hidden/"+id, mySubscriber);
}
```
?

Reply:
I cannot comment because now I do not understand how once should use your suggestion. Maybe now you should write a blog article ;-)

Reply:
@martin-g I don't really want to promote some dodgy non-streamline solution.

These are just examples of possible solutions for the problem:
https://github.com/andruhon/wicket-delayed-animation/tree/primitive-timeout-solution
https://github.com/andruhon/wicket-delayed-animation/tree/primitive-pubsub-solution
It has no wicket java, only wicket-ajax-jquery and a fake response in response.xml

Anyway I believe the solution should be javascripty, rather than something that has to be parsed. Maybe something like Wicket.Channel.pause() and Wicket.Channel.resume() if say we want to pause all ajax during this animation, or maybe some other methods *to be called from JS* to pause and notify particular request.

I'll think a little bit more.

Reply:
Hi @martin-g ! All right, I'm back with my dodgy experiments. I implemented the workaround to feed notify callback and parameters into global eval.

The change in wicket-ajax-jquery:
https://github.com/andruhon/wicket-delayed-animation/commit/f2aefa73902d89b9cfdae5f55fc2690c1b3282ac#diff-459f1fe1d12bc75eca45c5998e5dd29e

The branch in my quickstart project
https://github.com/andruhon/wicket-delayed-animation/tree/manual-notify-globalEval-closure

P.S.
Maybe I'm not very successful on expressing my thoughts. I do actually like the notify functionality itself, that's why I don't want to promote my non-streamline workarounds with timeouts. What I don't like is that it has to be added as a string `notify|` and then parsed. Why don't we add some kind of `notify="true"` attribute to the eval tag in response? What do you think?

Reply:
IF we're using a global reference anyway, we can just get rid of that ugly parameter and allow scripts to suspend the execution explicitly:

https://gist.github.com/svenmeier/bd4375ab26e330617a834e1d53d86556

Reply:
@svenmeier I like idea. The one thing I don't understand why the count is needed? As far as I know it's either waiting or proceeding, so I think it won't get to the request (with a suspend) unless It's not suspended.

@svenmeier, @martin-g please have a look at this implementation:
https://github.com/andruhon/wicket-delayed-animation/commit/e799a1dfbd24fd35e0701d452c95cbf51f732186
Here I simply disallow to call suspend when evaluation is not happening, also it doesn't allow to call suspend twice, so I think boolean in this case is enough.

I added two buttons with slow 2.5s suspended fade out, you can try quickly clicking them both.